### PR TITLE
Javadoc BigFormat

### PR DESCRIPTION
**Pull request by bereng**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

**Review by maoling — APPROVED** [review]
Looks great. +1(no binding)

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java [hunk 1]
```diff
@@ -66,7 +67,99 @@
...
- * Legacy bigtable format
+ * Legacy bigtable format. Components and approx lifecycle:
+ * <br>
+ * {@link SSTableFormat.Components}
+ * <br>
+ * {@link Components#ALL_COMPONENTS}
+ *  <ul>
+ *     <li>
+ *       {@link Components#SUMMARY}: When searching for a PK we go here for a first approximation on where to look in the index file. It is
+ *       a small sampling of the Index entries intended for a first fast search in-memory.
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java:78]: `for the first` probably? I am not a native speaker though.

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java [hunk 2]
```diff
@@ -66,7 +67,99 @@
 import static org.apache.cassandra.io.sstable.format.SSTableFormat.Components.DATA;
 
 /**
- * Legacy bigtable format
+ * Legacy bigtable format. Components and approx lifecycle:
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: can we use full for of this? - `approximate`? Later you have word "approximation".

Comment (smiklosovic) [inline src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java:70]: can we use full for of this? - `approximate`? Later you have word "approximation".

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java [hunk 3]
```diff
@@ -66,7 +67,99 @@
...
+ *     <li>
+ *       {@link Components#CRC}: CRC for the data
+ *       <p></p>
+ *       {@link org.apache.cassandra.io.util.ChecksummedSequentialWriter}
+ *       <br>
+ *       {@link org.apache.cassandra.io.util.ChecksumWriter}
+ *       <p></p>
+ *     </li>
+ *     <li>
+ *       {@link Components#TOC}: List of all the components for the sstable
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: Should not be "SSTable" ?

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java (hunk 2 above)
Comment (bereng) [reply src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: Done

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java (hunk 3 above)
Comment (bereng) [reply src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: Done

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java (hunk 1 above)
Comment (bereng) [reply src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java:78]: I wanted to say 'a' not 'the' :shrug:

Comment (smiklosovic) [reply src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java:78]: ok

File: src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java [hunk 4]
```diff
@@ -66,7 +67,99 @@
...
+ *       <p></p>
+ *     </li>
+ *     <li>
+ *       {@link Components#DATA}: The actual data/partitions file as an array or partitions. Each partition has the form:
+ *       <ul>
+ *       <li>A partition header</li>
+ *       <li>Maybe a static row</li>
+ *       <li>Rows or range tombstone</li>
+ *       </ul>
+ *       I.e. upon flush {@link Flushing.FlushRunnable#writeSortedContents()}
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: this has braces `()` but other links pointing to methods do no have it.

Comment (bereng) [reply src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java]: Done

**Pull request closed without merging**